holmes/core/tool_calling_llm.py

```diff
@@ -289,7 +295,12 @@ def _invoke_tool(
             logging.error(
                 f"Tool call to {tool_name} failed with an Exception", exc_info=True
             )
-            tool_response = f"Tool call failed: {e}"
+            tool_response = StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Tool call failed: {e}",
+                return_code=-1,
```

Comment:
Why is return code set to `-1`? `return_code` is optional so I'd expect it to be None for every single case where we don't have a valid one.